GW2EIEvtcParser/EncounterLogic/Raids/W8/UraTheSteamshrieker.cs

```diff
@@ -204,6 +205,57 @@ internal override void EIEvtcParse(ulong gw2Build, EvtcVersionEvent evtcVersion,
         }
     }
 
+    internal override List<PhaseData> GetPhases(ParsedEvtcLog log, bool requirePhases)
+    {
+        List<PhaseData> phases = GetInitialPhase(log);
+        SingleActor ura = Targets.FirstOrDefault(x => x.IsSpecies(TargetID.Ura)) ?? throw new MissingKeyActorsException("Ura not found"); ;
+        phases[0].AddTarget(ura);
+        if (!requirePhases)
+        {
+            return phases;
+        }
+
+        bool isCm = log.FightData.IsCM;
+        long start = log.FightData.FightStart;
+        long end = log.FightData.FightEnd;
+        var hpUpdates = ura.GetHealthUpdates(log);
+
+        var hp70 = hpUpdates.FirstOrDefault(x => x.Value < 70);
+        var hp40 = hpUpdates.FirstOrDefault(x => x.Value < 40);
+        var hp1 = log.CombatData.GetBuffData(Determined895).FirstOrDefault(x => x is BuffApplyEvent && x.To == ura.AgentItem);
+
+        // 100-70
+        var before70 = hp70.Value > 0 ? new PhaseData(start, hp70.Start, "100-70%") : new PhaseData(log.FightData.FightStart, end, "100-70%");
+        before70.AddTarget(ura);
+        phases.Add(before70);
+
+        // 70-40
+        if (hp70.Value > 0)
+        {
+            var after70 = new PhaseData(hp70.Start, Math.Min(hp40.Start, end), "70-40%");
+            after70.AddTarget(ura);
+            phases.Add(after70);
+        }
+
+        // 40 - 1 CM / 40 - 0 NM
+        if (hp40.Value > 0)
```

Comment:
I have not seen the CM yet, but I don't remember anything particular happening in normal mode at 40%, could you explain it to me please?

Reply:
The bubbles mechanic disappears at 70% and reactivates at 40%

Reply:
Any way to detect that in a more accurate manner? Otherwise I'd prefer if we had a general 70%-0%/1% phase and 70%-40% 40%-1%/0% be subphases of it